src/Microsoft.DotNet.Interactive.Http.Tests/HttpKernelTests.cs

```diff
@@ -347,6 +592,67 @@ public async Task url_in_embedded_expression_is_valid()
         request.RequestUri.AbsoluteUri.Should().Be("https://httpbin.org/anything");
     }
 
+    [Fact] 
+    public async Task incorrect_datetime_syntax_produces_error()
+    {
+        HttpRequestMessage request = null;
+        var handler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(handler);
+        using var kernel = new HttpKernel(client: client);
+
+        using var _ = new AssertionScope();
+
+        var code = """
+            POST https://api.example.com/comments HTTP/1.1
+            Content-Type: application/xml
+            
+            {
+                "local_custom_date": "{{$localDatetime 'YYYY-NN-DD}}"
+            }
+            """;
+
+        var result = await kernel.SendAsync(new SubmitCode(code));
+
+        result.Events.First().Equals(HttpDiagnostics.CannotResolveSymbol("{{$localDatetime 'YYYY-NN-DD}}"));
+
+    }
+
+
+    [Fact]
+    public async Task incorrect_datetime_produces_error()
+    {
+        HttpRequestMessage request = null;
+        var handler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(handler);
+        using var kernel = new HttpKernel(client: client);
+
+        using var _ = new AssertionScope();
+
+        var code = """
+            POST https://api.example.com/comments HTTP/1.1
+            Content-Type: application/xml
+            
+            {
+                "local_custom_date": "{{$localDatetime YYYY-MM-DD'}}"
+            }
+            """;
+
+        var result = await kernel.SendAsync(new SubmitCode(code));
+
+        result.Events.First().Equals(HttpDiagnostics.DateTimePatternMatchError("localDatetime YYYY-MM-DD"));
```

Comment:
Looks like not all of the new diagnostics in `HttpDiagnostics` have associated tests. Could we add (at least) one test for each error that the kernel catches and report to ensure that the corresponding code paths are covered?